Composite and compositepi

postvakje opened the pull request:
`composite(n)` returns the n-th composite number. 
`compositepi(n)` returns the number of positive composite numbers less than or equal to n.

postvakje committed:
Composite and compositepi

composite(n) returns the n-th composite number. compositepi(n) returns
the number of positive composite numbers less than or equal to n.

jtnydv25 commented:
A much better method for `composite(nth)` would be to use logarithmic integral, like in `prime(nth)`

postvakje committed:
rewrite composite() to be similar to prime()

postvakje commented:
Thank you for the suggestion.  I was experimenting with prime and primepi from version 1.0 which use the sieve.  I have now rewritten composite to be similar to the new prime().

postvakje committed:
adjusted some spacing

asmeurer commented on sympy/ntheory/generate.py:
```diff
@@ -741,3 +741,82 @@ def cycle_length(f, x0, nmax=None, values=False):
+
+
+def compositepi(n):
+    """ Return the number of positive composite numbers less than or equal to n.
```
Should note here too that 4 is the first composite

asmeurer commented on sympy/ntheory/generate.py:
```diff
@@ -741,3 +741,82 @@ def cycle_length(f, x0, nmax=None, values=False):
+        sympy.ntheory.primetest.isprime : Test if n is prime
+        primerange : Generate all primes in a given range
+        primepi : Return the number of primes less than or equal to n
+        prime : Return the nth prime
```
Mention conpositepi

postvakje committed:
added more details in docstring

added compositepi to example section of composite(nth)
added composite(1)=4 in docstring of compositepi

asmeurer commented:
Didn't notice this was finished. Please comment when you a ready for a PR to be reviewed again, as GitHub doesn't send notifications on new commits. 

This looks good to me. I'm merging.
